### PR TITLE
feat: add reka ai support

### LINKED ISSUE #302: [Provider] Reka AI
https://docs.reka.ai/http-api.html

### PR DESCRIPTION
**Pull request by meronogbai**
This adds support for [reka ai](https://www.reka.ai/).

**Related Issues:** (optional)
Closes https://github.com/Portkey-AI/gateway/issues/302

### CODE REVIEW

File: src/providers/reka-ai/chatComplete.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  top_p: {
+    param: 'runtime_top_p',
+  },
+  top_k: {
+    param: 'runtime_top_k',
+  },
+  stop: {
+    param: 'stop_words',
+  },
+
```
Comment (meronogbai) [inline src/providers/reka-ai/chatComplete.ts:45]: Their [docs](https://docs.reka.ai/http-api.html#generation) have more options, but I wasn't sure how to map them here. What do you think would be the best mapping?

File: src/providers/reka-ai/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { ProviderConfigs } from '../types';
+import RekaAIApiConfig from './api';
+import {
+  RekaAIChatCompleteConfig,
+  RekaAIChatCompleteResponseTransform,
+} from './chatComplete';
+
+const RekaAIConfig: ProviderConfigs = {
+  chatComplete: RekaAIChatCompleteConfig,
```
Comment (meronogbai) [inline src/providers/reka-ai/index.ts:9]: They don't seem to have a streaming endpoint so I only added chat completion support.

File: src/providers/reka-ai/chatComplete.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  },
+  top_p: {
+    param: 'runtime_top_p',
+  },
+  top_k: {
+    param: 'runtime_top_k',
+  },
+  stop: {
+    param: 'stop_words',
+  },
```
Comment (meronogbai) [inline src/providers/reka-ai/chatComplete.ts]: Their [docs](https://docs.reka.ai/http-api.html#generation) have more options (use_search_engine, retrieval_dataset, etc), but I don't know how to map them to openai's options.  What do you think would be the best mapping?

Comment (VisargD) [reply src/providers/reka-ai/chatComplete.ts]: For all the params which do not have an openai equivalent param, you can add them as it is in the mapping

Comment (meronogbai) [reply src/providers/reka-ai/chatComplete.ts]: Done!

File: src/providers/reka-ai/chatComplete.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+            addMessage({
+              type: currentType,
+              text: item.text || '',
+              media_url: item.image_url?.url
+            });
+          });
+        }
+      });
+
+      if (messages[0].type !== 'human') {
```
Comment (meronogbai) [inline src/providers/reka-ai/chatComplete.ts:79]: First item must be human. 

See https://docs.reka.ai/guides/001-chat.html#conversation-history

File: src/providers/reka-ai/chatComplete.ts [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+          const placeholder: RekaMessageItem = {
+            type: type === 'human' ? 'model' : 'human',
+            text: 'Placeholder for alternation'
+          };
+          media_url ? messages.unshift(placeholder) : messages.push(placeholder);
+        }
+
+        // NOTE: image need to be first
+        media_url ? messages.unshift(newMessage) : messages.push(newMessage);
+        lastType = type;
```
Comment (meronogbai) [inline src/providers/reka-ai/chatComplete.ts:59]: There can only be one image in the conversation history. Moreover, we can only use images on the first turn.

See https://docs.reka.ai/guides/002-image-video-audio-chat.html#multiple-turns

File: src/providers/reka-ai/index.ts (hunk 2 above)
Comment (VisargD) [reply src/providers/reka-ai/index.ts:9]: Acknowledged. We have a json to stream function but we can use that in another PR.

File: src/providers/reka-ai/chatComplete.ts (hunk 4 above)
Comment (VisargD) [reply src/providers/reka-ai/chatComplete.ts:79]: Acknowledged

File: src/providers/reka-ai/chatComplete.ts (hunk 5 above)
Comment (VisargD) [reply src/providers/reka-ai/chatComplete.ts:59]: Acknowledged

**Comment by VisargD** [comment]
Hey. Thanks for the PR. Whenever possible, can you please resolve the conflicts.

**Comment by VisargD** [comment]
Hey @meronogbai - Please look into the merge conflicts whenever possible.

**Comment by meronogbai** [comment]
I’ll tale a look tonight.

**Comment by meronogbai** [comment]
Done 🫡

**Review by VisargD — APPROVED** [review]

**Pull request merged**